opm/autodiff/ParallelOverlappingILU0.hpp

```diff
@@ -393,6 +420,22 @@ class ParallelOverlappingILU0
 protected:
     void init( const Matrix& A, const int iluIteration )
     {
+        // (For older DUNE versions the communicator might be
+        // invalid if redistribution in AMG happened on the coarset level.
+        // Therefore we check for nonzero size
+        if ( comm_ && comm_->communicator().size()<=0 )
+        {
+            if ( A.N() > 0 )
+            {
+                OPM_THROW(std::logic_error, "Expected a matrix with zero rows for an invalid communicator.");
```

Comment:
This requires including opm/common/ErrorMacros.hpp.

There seems to be a commented-out OPM_THROW in apply() around line 360 that could have been disabled for the same reason? Right now it uses std::abort() instead which seems a little harsh.